kaldb/src/main/java/com/slack/kaldb/logstore/opensearch/OpenSearchAdapter.java

```diff
@@ -96,10 +97,15 @@ public class OpenSearchAdapter {
 
   private final QueryShardContext queryShardContext;
 
-  private final MapperService mapperService;
+  @VisibleForTesting protected final MapperService mapperService;
```

Comment:
One thought here is just to change the `tryRegisterField` to not be static, and don't pass the mapper service in each call. That'd actually cleanup the `reloadSchema` method a bit, and then you wouldn't need to make the mapper service visible. You could just have the `tryRegisterField` as a protected.